### PR TITLE
feat(connectors): add authorization section to http webhook connector

### PR DESCRIPTION
**Pull request by markfarkas-camunda**
## Description

JWT authorization was added as a feature to the webhook connector, so the documentation had to be adjusted.

## When should this change go live?

After the [related task](https://github.com/camunda/connectors-bundle/issues/629) is completed and https://github.com/camunda/connectors-bundle/pull/756 is merged

- [x] This change is not yet live and should not be merged until  [related task](https://github.com/camunda/connectors-bundle/issues/629) is completed and https://github.com/camunda/connectors-bundle/pull/756 is merged
- [x] There is no urgency with this change.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/connectors/protocol/http-webhook.md [hunk 1]
```diff
@@ -27,14 +27,22 @@ Please refer to the [update guide](/guides/update-guide/connectors/060-to-070.md
...
 
-3. Configure **Activation Condition**. For example, given external caller triggers a webhook endpoint with the body `{"id": 1, "status": "OK"}`, the **Activation Condition** value might look like `=(request.body.status = "OK")`. Leave this field empty to trigger your webhook every time.
-4. Use **Variable Mapping** to map specific fields from the request into process variables using [FEEL](/components/modeler/feel/what-is-feel.md).
+3. Configure [JWT authorization](https://jwt.io/) if required in the **Authorization** section. The token should be in the _Authorization_ header of the request in the format of Bearer {JWT_TOKEN}
+
+- Set JWK url which is used as a well-known public url to fetch the JWKs.
+- Set JWT role property expression which will be evaluated against the content of the JWT to extract the list of roles.
+- Set required roles which will be used to validate if the JWT contains all required roles.
+
+![HTTP Webhook jwt](../img/connectors-http-webhook-jwt.png)
```
Comment (sbuettner) [inline docs/components/connectors/protocol/http-webhook.md:42]: As stated here https://github.com/camunda/team-connectors/issues/448#issuecomment-1634427431 lets remove the screenshots.

Please provide an example of the role example under a linkable session (that can be linked to from the element template) that shows an example JWT payload and the respective expression that includes the role extraction.

File: versioned_docs/version-8.1/components/connectors/protocol/http-webhook.md [hunk 2]
```diff
@@ -27,15 +27,22 @@ Please refer to the [update guide](/guides/update-guide/connectors/060-to-070.md
...
-3. Configure **Activation Condition**. For example, given external caller triggers a webhook endpoint with the body `{"id": 1, "status": "OK"}`, the **Activation Condition** value might look like `=(request.body.status = "OK")`. Leave this field empty to trigger your webhook every time.
-4. Use **Result Variable** to store the response in a process variable. For example, `myResultVariable`.
-5. Use **Result Expression** to map specific fields from the response into process variables using [FEEL](/components/modeler/feel/what-is-feel.md).
+3. Configure [JWT authorization](https://jwt.io/) if required in the **Authorization** section. The token should be in the _Authorization_ header of the request in the format of Bearer {JWT_TOKEN}
+
+- Set JWK url which is used as a well-known public url to fetch the JWKs.
+- Set JWT role property expression which will be evaluated against the content of the JWT to extract the list of roles.
+- Set required roles which will be used to validate if the JWT contains all required roles.
+
+![HTTP Webhook jwt](../img/connectors-http-webhook-jwt.png)
```
Comment (sbuettner) [inline versioned_docs/version-8.1/components/connectors/protocol/http-webhook.md:42]: Is webhook available in `8.1` at all 😄 ?

Comment (markfarkas-camunda) [reply versioned_docs/version-8.1/components/connectors/protocol/http-webhook.md:42]: We had the documentation for 8.1, so I hope so. I think inbound connectors are available for 8.1, so I guess webhook as well?

File: docs/components/connectors/protocol/http-webhook.md (hunk 1 above)
Comment (markfarkas-camunda) [reply docs/components/connectors/protocol/http-webhook.md]: Good idea, added the example and removed screenshots

**Review by sbuettner — APPROVED** [review]
Great work!

**Comment by christinaausley** [comment]
Let me know when you are ready for a technical review here 👍

**Comment by markfarkas-camunda** [comment]
> Let me know when you are ready for a technical review here 👍

@christinaausley sorry for the late reply. Yes, we are ready for a technical review here

File: docs/components/connectors/protocol/http-webhook.md [hunk 3]
```diff
@@ -27,14 +27,32 @@ Please refer to the [update guide](/guides/update-guide/connectors/060-to-070.md
...
 
 - Set the HMAC shared secret key which is used to calculate the message hash. The value is defined by a webhook administrator.
 - Set the HMAC header whose value contains an encrypted hash message. The exact value is provided by the external caller.
 - Select HMAC hash algorithm. The exact value is provided by the external caller.
 
-3. Configure **Activation Condition**. For example, given external caller triggers a webhook endpoint with the body `{"id": 1, "status": "OK"}`, the **Activation Condition** value might look like `=(request.body.status = "OK")`. Leave this field empty to trigger your webhook every time.
-4. Use **Variable Mapping** to map specific fields from the request into process variables using [FEEL](/components/modeler/feel/what-is-feel.md).
+3. Configure authorization if required in the **Authorization** section. The HTTP Webhook Connector supports the following authorization methods:
+
+- **Basic** - The incoming requests must contain an `Authorization` header that contains the word Basic word followed by a space and a base64-encoded string username:password.
```
Comment (christinaausley) [inline docs/components/connectors/protocol/http-webhook.md:38]: Do you mean "That contains the word `Basic`" here?

File: docs/components/connectors/protocol/http-webhook.md [hunk 4]
```diff
@@ -27,14 +27,32 @@ Please refer to the [update guide](/guides/update-guide/connectors/060-to-070.md
...
+
+  - Set the **Username** and **Password** properties which will be used to validate the incoming requests.
+  - Please provide the values in plain text, not base64-encoded.
+
+- **API Key** - The API key can be provided anywhere in the request, for example, in the `Authorization` header or in the request body.
+
+  - Set the **API Key** property to the expected value of the API key.
+  - Set the **API Key locator** property that will be evaluated against the incoming request to extract the API key. [See the example](#how-to-configure-api-key-authorization).
+
+- **[JWT authorization](https://jwt.io/)** - The token should be in the _Authorization_ header of the request in the format of Bearer {JWT_TOKEN}
```
Comment (christinaausley) [inline docs/components/connectors/protocol/http-webhook.md:48]: Just need a period here to complete the sentence.

**Review by christinaausley — DISMISSED** [review]
Technical review in latest commit, and one comment for clarification 👍

File: docs/components/connectors/protocol/http-webhook.md (hunk 3 above)
Comment (chillleader) [reply docs/components/connectors/protocol/http-webhook.md:38]: Exactly! Good catch

**Review by christinaausley — APPROVED** [review]

**Comment by sbuettner** [comment]
@markfarkas-camunda @chillleader Can we merge this?

**Review by sbuettner — APPROVED** [review]

**Comment by chillleader** [comment]
@sbuettner we are holding it until this functionality is released, as the changes also apply to already released docs for versions 8.1 and 8.2.

**Comment by akeller** [comment]
Based on the description, both the related task and PR are completed. However, there are merge conflicts here. 

Are you expecting this to go out in the alpha tomorrow? If so, please make any changes in your morning so we can review in the US morning and do docs release to support the alpha.

**Comment by markfarkas-camunda** [comment]
> Based on the description, both the related task and PR are completed. However, there are merge conflicts here.
> 
> Are you expecting this to go out in the alpha tomorrow? If so, please make any changes in your morning so we can review in the US morning and do docs release to support the alpha.

@akeller Merge conflicts resolved, please review it and if everything is okay, merge it, to release this in the alpha today.

**Pull request merged**